include/wil/winrt.h

```diff
@@ -994,6 +1000,42 @@ namespace wil
     };
 #pragma endregion
 
+#if defined(__WI_HAS_STD_VECTOR)
+    /** Converts WinRT vectors to std::vector by requesting the collection's
+    data in chunks. This can be more efficient in terms of IPC cost than
+    iteratively processing the collection.
+    ~~~
+    ComPtr<IVector<IPropertyValue*>> values = GetValues();
+    std::vector<ComPtr<IPropertyValue>> allData = wil::to_vector(values);
+    for (ComPtr<IPropertyValue> const& item : allData)
+    {
+        // use item
+    }
+    Can be used for ABI::Windows::Foundation::Collections::IVector<T> and
+    ABI::Windows::Foundation::Collections::IVectorView<T>
+    */
+    template<typename VectorType> auto to_vector(VectorType* src)
+    {
+        using TResult = typename details::MapVectorResultType<VectorType>::type;
+        using TSmart = typename details::MapToSmartType<TResult>::type;
+        static_assert(sizeof(TResult) == sizeof(TSmart), "result and smart sizes are different");
+        std::vector<TSmart> output;

```

Comment:
Sadly there's (not yet) a vector-like wrapper for the wil::unique_foo_ptr<T[]> types that supports resize/enlarge/append. Making the nothrow non-stl version of this was going to be a lot of code.  I'm open to writing that, though, if folks think it's useful.

Reply:
Hmm, see also #242 that is probably a prerequisite to making this one easy. Neither `wistd::unique_ptr<T[]>` (no length member) nor `wil::unique_cotaskmem_array_ptr<T>` (no call to dtor) are directly suitable.